cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/TimePartitionedFileSetDataset.java

```diff
@@ -149,6 +149,7 @@ public TimePartitionOutput getPartitionOutput(long time) {
         "Output is not supported for external time-partitioned file set '" + spec.getName() + "'");
     }
     PartitionKey key = partitionKeyForTime(time);
+    assertNotExists(key);
```

Comment:
`PartitionedFileSet#getOutputPartition` now requires a transaction.
Seems like a pretty big bug, that we previously gave users a PartitionOutput object, for a partition that already exists, and the only way to avoid that is requiring that `PartitionedFileSet#getOutputPartition` is executed within a transaction, so that we can check if the partition already exists or not, before returning from it.
I'll add this note to the release note for this issue.